echopop/nwfsc_feat/spatial.py

```diff
@@ -0,0 +1,639 @@
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
+from lmfit import Minimizer, Parameters
+from scipy import interpolate
+
+from ..spatial.variogram import variogram
+
+
+def standardize_coordinates(
+    data_df: pd.DataFrame,
+    longitude_offset: float = 0.0,
+    latitude_offset: float = 0.0,
+    reference_df: Optional[pd.DataFrame] = None,
+    delta_longitude: Optional[float] = None,
+    delta_latitude: Optional[float] = None,
+) -> Tuple[pd.DataFrame, Union[float, None], Union[float, None]]:
+    """
+    Standardize the longitude and latitude coordinates of a dataset.
+
+    Parameters
+    ----------
+    data_df : pd.DataFrame
+        DataFrame with longitude and latitude coordinates
+    longitude_offset : float, default=0.
+        Offset to apply to the longitude coordinates
+    latitude_offset : float, default=0.
+        Offset to apply to the latitude coordinates
+    reference_df : pd.DataFrame, optional
+        Reference DataFrame with longitude and latitude coordinates for interpolation that is
+        used as an additional offset to longitude
+    delta_longitude : float, optional
+        Total longitudinal distance (degrees) used for standardizing coordinates
+    delta_latitude : float, optional
+        Total latitudinal distance (degrees) used for standardizing coordinates
+
+    Returns
+    -------
+    pd.DataFrame
+        DataFrame with the new standardized coordinates 'x' and 'y'.
+    float or None
+        Total longitudinal distance (degrees) used for standardizing coordinates that can be used
+        for the transformation of other georeferenced datasets.
+    float or None
+        Total latitudinal distance (degrees) used for standardizing coordinates that can be used
+        for the transformation of other georeferenced datasets.
```

Comment:
The docstrings here reads messy - are you talking about the transformation or standardization step?